satpy/readers/viirs_atms_sdr_base.py

```diff
@@ -0,0 +1,366 @@
+
+
+class JPSS_SDR_FileHandler(HDF5FileHandler):
+    """Base class for reading JPSS VIIRS & ATMS SDR HDF5 Files."""
+
+    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+        """Initialize file handler."""
+        super().__init__(filename, filename_info, filetype_info, **kwargs)
+
+    def _parse_datetime(self, datestr, timestr):
+        try:
+            datetime_str = datestr + timestr
+        except TypeError:
+            datetime_str = (str(datestr.data.compute().astype(str)) +
+                            str(timestr.data.compute().astype(str)))
+
+        time_val = datetime.strptime(datetime_str, '%Y%m%d%H%M%S.%fZ')
+        if abs(time_val - NO_DATE) < EPSILON_TIME:
+            # catch rare case when SDR files have incorrect date
+            raise ValueError("Datetime invalid {}".format(time_val))
+        return time_val
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        dataset_group = DATASET_KEYS[self.datasets[0]]
+        default_start_date = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/AggregateBeginningDate'
+        default_start_time = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/AggregateBeginningTime'
+        date_var_path = self.filetype_info.get('start_date', default_start_date).format(dataset_group=dataset_group)
+        time_var_path = self.filetype_info.get('start_time', default_start_time).format(dataset_group=dataset_group)
+        return self._parse_datetime(self[date_var_path], self[time_var_path])
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        dataset_group = DATASET_KEYS[self.datasets[0]]
+        default_end_date = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/AggregateEndingDate'
+        default_end_time = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/AggregateEndingTime'
+        date_var_path = self.filetype_info.get('end_date', default_end_date).format(dataset_group=dataset_group)
+        time_var_path = self.filetype_info.get('end_time', default_end_time).format(dataset_group=dataset_group)
+        return self._parse_datetime(self[date_var_path], self[time_var_path])
+
+    @property
+    def start_orbit_number(self):
+        """Get start orbit number."""
+        dataset_group = DATASET_KEYS[self.datasets[0]]
+        default = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/AggregateBeginningOrbitNumber'
+        start_orbit_path = self.filetype_info.get('start_orbit', default).format(dataset_group=dataset_group)
+        return int(self[start_orbit_path])
+
+    @property
+    def end_orbit_number(self):
+        """Get end orbit number."""
+        dataset_group = DATASET_KEYS[self.datasets[0]]
+        default = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/AggregateEndingOrbitNumber'
+        end_orbit_path = self.filetype_info.get('end_orbit', default).format(dataset_group=dataset_group)
+        return int(self[end_orbit_path])
```

Comment:
```suggestion
    @property
    def end_orbit_number(self):
        """Get end orbit number."""
        end_orbit_path = self._get_aggr_path("end_orbit", "AggregateEndingOrbitNumber")
        return int(self[end_orbit_path])
        
    def _get_aggr_path(self, fileinfo_key, aggr_default):
        dataset_group = DATASET_KEYS[self.datasets[0]]
        default = 'Data_Products/{dataset_group}/{dataset_group}_Aggr/attr/' + aggr_default
        return self.filetype_info.get(fileinfo_key, default).format(dataset_group=dataset_group)
```

We could introduce this new method and reduce the code duplication in the four previous methods

Reply:
You can use `_get_aggr_path` in the `start_orbit_number`, `start_time` and `end_time` methods too.

Reply:
Fixed!